modules/ffmpeg/producer/ffmpeg_producer.cpp

```diff
@@ -405,74 +408,77 @@ struct ffmpeg_producer : public core::frame_producer_base
+		return muxer_->calc_nb_frames(nb_frames);
 	}
 
 	uint32_t file_nb_frames() const
 	{
-		uint32_t file_nb_frames = 0;
-		file_nb_frames = std::max(file_nb_frames, video_decoder_ ? video_decoder_->nb_frames() : 0);
-		return file_nb_frames;
+		return video_decoder_ ? video_decoder_->nb_frames() : 0;
 	}
 
 	std::future<std::wstring> call(const std::vector<std::wstring>& params) override
 	{
-		static const boost::wregex loop_exp(LR"(LOOP\s*(?<VALUE>\d?)?)", boost::regex::icase);
-		static const boost::wregex seek_exp(LR"(SEEK\s+(?<VALUE>(\+|-)?\d+)(\s+(?<WHENCE>REL|END))?)", boost::regex::icase);
-		static const boost::wregex length_exp(LR"(LENGTH\s+(?<VALUE>\d+)?)", boost::regex::icase);
-		static const boost::wregex start_exp(LR"(START\s+(?<VALUE>\d+)?)", boost::regex::icase);
-
-		auto param = boost::algorithm::join(params, L" ");
-
 		std::wstring result;
 
-		boost::wsmatch what;
-		if(boost::regex_match(param, what, loop_exp))
-		{
-			auto value = what["VALUE"].str();
-			if (!value.empty())
-				input_.loop(boost::lexical_cast<bool>(value));
-			result = boost::lexical_cast<std::wstring>(input_.loop());
-		}
-		else if(boost::regex_match(param, what, seek_exp))
+		if(params.size())
 		{
-			auto value = boost::lexical_cast<int64_t>(what["VALUE"].str());
-			auto whence = what["WHENCE"].str();
-			auto total = file_nb_frames();
+			std::wstring cmd = params.at(0), value;
+			if(params.size() > 1) value = params.at(1);
 
-			if(boost::iequals(whence, L"REL"))
-				value = file_frame_number() + value;
-			else if(boost::iequals(whence, L"END"))
-				value = total - value;
+			if(boost::iequals(cmd, L"loop"))
+			{
+				if(value.size())
+					input_.loop(boost::lexical_cast<bool>(value));
+				result = boost::lexical_cast<std::wstring>(input_.loop());
+			}
+			else if(boost::iequals(cmd, L"in") || boost::iequals(cmd, L"start"))
+			{
+				if(value.size())
+					input_.in(boost::lexical_cast<uint32_t>(value));
+				result = boost::lexical_cast<std::wstring>(input_.in());
+			}
+			else if(boost::iequals(cmd, L"out"))
+			{
+				if(value.size())
+					input_.out(boost::lexical_cast<uint32_t>(value));
+				result = boost::lexical_cast<std::wstring>(input_.out());
+			}
+			else if(boost::iequals(cmd, L"length"))
+			{
+				if(value.size())
+					input_.length(boost::lexical_cast<uint32_t>(value));
+				result = boost::lexical_cast<std::wstring>(input_.length());
+			}
+			else if(boost::iequals(cmd, L"seek") && value.size())
+			{
+				auto nb_frames = file_nb_frames();
+
+				int64_t seek;
+				if(boost::iequals(value, L"rel"))
+					seek = file_frame_number();
+				else if(boost::iequals(value, L"in"))
+					seek = input_.in();
+				else if(boost::iequals(value, L"out"))
+					seek = input_.out();
+				else if(boost::iequals(value, L"end"))
+					seek = nb_frames;
+				else
+					seek = boost::lexical_cast<int64_t>(value);
 
-			if(value < 0)
-				value = 0;
-			else if(value >= total)
-				value = total - 1;
+				if(params.size() > 2)
```

Comment:
Is it intended to be optional? If no, then it is better to let the out of range exception propagate here as it is handled by https://github.com/CasparCG/Server/blob/2.1.0/protocol/amcp/AMCPCommandQueue.cpp#L129 instead